### ISSUE TITLE
COLLATE ci on embedded collection

### ISSUE THREAD

**Issue by gongdo** [orientechnologies/orientdb#4851]
I'm using v2.1, I tried to set `COLLATE ci` on the `EMBEDDEDLIST string` property.
It was a mistake thought, its behavior was quite wired, look at this:
```
create class test extends V
// assume that the cluster is #12
create property test.categories EMBEDDEDLIST string
create vertex test set categories=['a','b'] // #12:0

select from test where 'a' in categories
// [OK] record #12:0 found 

alter property test.categories COLLATE ci  // <-- mistake, but it applied without error
create vertex test set categories=['Math','English']  // #12:1
create vertex test set categories=['a','b','c']  // #12:2

select from test where 'a' in categories
// [OK] record #12:0, #12:2 found 

select from test where 'Math' in categories
// [NOT OK] no records. expected #12:1.

alter property test.categories COLLATE null  // <-- remove the cause of problem.
select from test where 'Math' in categories
// [OK] record #12:1 found as expected
```

It seems that to set `COLLATE ci` on the `EMBEDDEDLIST`(or `SET`), makes `IN` operator dysfunctional. It looks like a bug which is pretty hard to figure out.

**Comment by obi1kenobi** [comment]
This is still broken in 2.1.5 and is really annoying. Any ETA on when it will be fixed?

**Comment by obi1kenobi** [comment]
Hey guys, I was wondering if this issue can get any love in the newest 2.1.x release by any chance. This is causing lots of :cry: and :disappointed: for my project.

**Comment by luigidellaquila** [comment]
Hi guys, I'm taking a look at it, I'll give you a feedback in short

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi guys,

I have a fix for this (that is also an enhancement, because it extends collate strategies to collections).
I'm running all the tests, I'll push the fix as soon as all is green

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Fix pushed on 2.1.x branch, now I'm porting it to 2.2

**Comment by luigidellaquila** [comment]
Ok, fixed, I'm closing this issue, please reopen if you experience any problems

Thanks

Luigi
